applications/MappingApplication/custom_mappers/interpolative_mapper_base.h

```diff
@@ -470,6 +486,80 @@ class KRATOS_API(MAPPING_APPLICATION) InterpolativeMapperBase : public Mapper<TS
         KRATOS_CATCH("");
     }
 
+    // Map expressions to variables with `double` value type, or value types that consist `double`s
+    Expression::Pointer MapInternal(Expression::Pointer pOriginExpression,
+                                    Kratos::Flags MappingOptions)
+    {
+        // Disable options that make no sense for expressions
+        // => Mapper has no control over where the origin data is fetched from
+        //    or where the destination expression gets written.
+        KRATOS_ERROR_IF(MappingOptions.Is(MapperFlags::FROM_NON_HISTORICAL)) << "The FROM_NON_HISTORICAL is disabled when mapping expressions";
+        KRATOS_ERROR_IF(MappingOptions.Is(MapperFlags::TO_NON_HISTORICAL)) << "The TO_NON_HISTORICAL is disabled when mapping expressions";
+        KRATOS_ERROR_IF(MappingOptions.Is(MapperFlags::ADD_VALUES)) << "The ADD_VALUES is disabled when mapping expressions";
+
+        KRATOS_ERROR_IF(MappingOptions.Is(MapperFlags::USE_TRANSPOSE)) << "Transpose mapping with expressions is not implemented";
+
+        KRATOS_TRY;
+
+        const std::size_t origin_size = pOriginExpression->size();
+        const std::size_t origin_node_count = mrModelPartOrigin.GetCommunicator().LocalMesh().NumberOfNodes();
+        const std::size_t destination_node_count = mrModelPartDestination.GetCommunicator().LocalMesh().NumberOfNodes();
+        KRATOS_ERROR_IF_NOT(((origin_size == 0) == (origin_node_count == 0)) && (origin_size == 0 ? true : !bool(origin_size % origin_node_count)))
+            << "Origin expression size (" << origin_size
+            << ") must be a multiple of the number of local nodes in the origin model part (" << mrModelPartOrigin.GetCommunicator().LocalMesh().NumberOfNodes() << ')';
+
+        // Get the pointer to the first item in the local vector
+        // Sadly, this isn't starightforward because operator[] in Trilinos vector types
+        // returns a double*& instead of a double&, so both cases must be handled.
+        using ComponentType = typename TSparseSpace::DataType;
+        using VectorSubscriptType = decltype(std::declval<typename TSparseSpace::VectorType>()[0]);
+        const auto get_vector_begin = [](InterfaceVectorContainerType& rVector) -> std::optional<ComponentType*> {
+            auto p_begin = std::optional<ComponentType*>();
+            if constexpr (std::is_same_v<VectorSubscriptType,ComponentType*&>) {
+                p_begin = rVector.GetVector()[0];
+            } else if constexpr (std::is_same_v<VectorSubscriptType,ComponentType&>) {
+                p_begin = &rVector.GetVector()[0];
+            }
+            return p_begin;
+        };
```

Comment:
the unified treatment of ublas and trilinos vectors was the origin of the `InterfaceVectorContainer`. Could this be solved better by moving this part there? Maybe then the nasty tricks would not be required?